### PR TITLE
Update edu benefits review page design

### PR DESCRIPTION
**Pull request by jbalboni**
Related to #3267 and #4067.

This involved changing a lot of code. The main challenge was that for the pages with growable tables the design doesn't have clearly separated review and edit states. We want to edit the growable table rows in place. This meant I needed to ditch the separate review/fields components used for most pages and handle both states in one component.

The high level changes are:

- Changed collapsible panel component to take a list of pages and render those inside the accordion
- Updated collapsible panel to pass review/edit states and hooks to the component its rendering if a review component was not passed. I'm doing this so that we can still use the standard Review and Edit layouts we had before, but do more custom ones for the three pages that need it.
- Added a bunch of options to GrowableTable to support the above.

screen shot 2016-11-22 at 1 54 11 pm
screen shot 2016-11-22 at 1 54 17 pm
screen shot 2016-11-22 at 1 55 28 pm
screen shot 2016-11-22 at 1 55 57 pm
screen shot 2016-11-22 at 1 56 13 pm
screen shot 2016-11-22 at 1 56 19 pm

### CODE REVIEW

**Comment by cweber** [comment]
Yay! That feel so much better to use.

To add:
- Scroll to top of section on expand or after updating

To fix:
- updating a card is causing a seemingly unpredictable scroll pattern. I can't tell what it's trying to scroll to, but it's odd.

**Comment by goldenmeanie** [comment]
apply_for_education_benefits__vets_gov

Clicking add another for a growable table with children scrolls half-way into form. Probably same/similar issue @cweber mentions for edit.

File: src/js/edu-benefits/components/education-history/EducationHistoryFields.jsx [hunk 1]
```diff
@@ -38,33 +65,44 @@ export default class EducationHistoryFields extends React.Component {
...
+        {this.props.editing
+          ? formView
+          : <EducationHistoryReview data={this.props.data} onEdit={this.props.onEdit}/>}
+      </div>
+      <div className="form-review-panel-page">
+        <div className="form-review-panel-page-header-row edu-growable-review-header">
+          <h5 className="form-review-panel-page-header">Education after high school</h5>
+          <button
+              className="edit-btn primary-outline"
+              onClick={this.addAnotherPeriod}><i className="fa before-text fa-pencil"></i>Add Another</button>
```
Comment (akainic) [inline src/js/edu-benefits/components/education-history/EducationHistoryFields.jsx]: The addAnother button appears in review mode?

File: src/js/edu-benefits/reducers/uiState/index.js [hunk 2]
```diff
@@ -21,83 +20,67 @@ const ui = {
   },
   pages: {
     '/introduction': {
-      complete: false,
-      verified: false,
+      reviewEdit: false,
```
Comment (akainic) [inline src/js/edu-benefits/reducers/uiState/index.js]: The name of this property is a little confusing to me. To me on the review page there are 2 states: `review` or `edit`. I think I know what you're trying to get at here, is the state in `edit` on the review page? Maybe call this `editOnReview` on something like that instead?

File: src/js/common/components/form-elements/GrowableTable.jsx [hunk 3]
```diff
@@ -212,12 +248,22 @@ GrowableTable.propTypes = {
   isValidSection: React.PropTypes.func.isRequired,
   addNewMessage: React.PropTypes.string,
   minimumRows: React.PropTypes.number,
-  rowTitle: React.PropTypes.string
+  rowTitle: React.PropTypes.string,
```
Comment (akainic) [inline src/js/common/components/form-elements/GrowableTable.jsx]: Pretty sure HCA now uses the common `GrowableTable` component, so these changes may also affect HCA. Need to make sure everything works there as well.

**Review by akainic — COMMENTED** [review]
This is huge! Changes look good overall. A little hard to review everything. In general I think some comments in the code to explain some of the new props and how they're used could be really helpful. Because this is adding so much more complexity I want to make sure it's understandable to future users of these components.

File: src/js/edu-benefits/components/education-history/EducationHistoryFields.jsx (hunk 1 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/education-history/EducationHistoryFields.jsx]: Yeah, if you're on the review page it always appears.

File: src/js/edu-benefits/reducers/uiState/index.js (hunk 2 above)
Comment (jbalboni) [inline src/js/edu-benefits/reducers/uiState/index.js]: `editOnReview` makes sense to me. I'll use that.

File: src/js/common/components/form-elements/GrowableTable.jsx (hunk 3 above)
Comment (jbalboni) [inline src/js/common/components/form-elements/GrowableTable.jsx]: Just checked and it seems fine. I made a minor tweak to the button style and fixed a validation issue on the review page that I noticed, though.

**Pull request merged**